db/views/reporting_patient_states_per_month_v03.sql

```diff
@@ -0,0 +1,173 @@
+    assigned_facility.state_slug as assigned_state_slug,
+    assigned_facility.state_region_id as assigned_state_region_id,
+    assigned_facility.organization_slug as assigned_organization_slug,
+    assigned_facility.organization_region_id as assigned_organization_region_id,
+
+    ------------------------------------------------------------
+    -- information on registration facility and parent regions
+    p.registration_facility_id AS registration_facility_id,
+    registration_facility.facility_size as registration_facility_size,
+    registration_facility.facility_type as registration_facility_type,
+    registration_facility.facility_region_slug as registration_facility_slug,
+    registration_facility.facility_region_id as registration_facility_region_id,
+    registration_facility.block_slug as registration_block_slug,
+    registration_facility.block_region_id as registration_block_region_id,
+    registration_facility.district_slug as registration_district_slug,
+    registration_facility.district_region_id as registration_district_region_id,
+    registration_facility.state_slug as registration_state_slug,
+    registration_facility.state_region_id as registration_state_region_id,
+    registration_facility.organization_slug as registration_organization_slug,
+    registration_facility.organization_region_id as registration_organization_region_id,
+
+    ------------------------------------------------------------
+    -- details of the visit: latest BP, encounter, prescription drug and appointment
+    bps.blood_pressure_id as blood_pressure_id,
+    bps.blood_pressure_facility_id AS bp_facility_id,
+    bps.blood_pressure_recorded_at AS bp_recorded_at,
+    bps.systolic,
+    bps.diastolic,
+
+    visits.encounter_id AS encounter_id,
+    visits.encounter_facility_id AS encounter_facility_id,
+    visits.encounter_recorded_at AS encounter_recorded_at,
+
+    visits.prescription_drug_id AS prescription_drug_id,
+    visits.prescription_drug_facility_id AS prescription_drug_facility_id,
+    visits.prescription_drug_recorded_at AS prescription_drug_recorded_at,
+
+    visits.appointment_id AS appointment_id,
+    visits.appointment_creation_facility_id AS appointment_creation_facility_id,
+    visits.appointment_recorded_at AS appointment_recorded_at,
+
+    ------------------------------------------------------------
+    -- relative time calculations
+
+    (cal.year - DATE_PART('year', p.recorded_at AT TIME ZONE 'UTC' AT TIME ZONE (SELECT current_setting('TIMEZONE')))) * 12 +
+    (cal.month - DATE_PART('month', p.recorded_at AT TIME ZONE 'UTC' AT TIME ZONE (SELECT current_setting('TIMEZONE'))))
+    AS months_since_registration,
+
+    (cal.year - DATE_PART('year', p.recorded_at AT TIME ZONE 'UTC' AT TIME ZONE (SELECT current_setting('TIMEZONE')))) * 4 +
+    (cal.month - DATE_PART('quarter', p.recorded_at AT TIME ZONE 'UTC' AT TIME ZONE (SELECT current_setting('TIMEZONE'))))
+    AS quarters_since_registration,
```

Comment:
The repetition here makes me wonder whether `_since_registration` is required in the BPs and Visits tables. 🤔 

If we're not sure, let's leave it as is.

Reply:
It might be useful if we're using those tables in ad-hoc reports. It's useful to leave in there, ya.